qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/LocationManagementSteps.java

```diff
@@ -0,0 +1,288 @@
+    private ManageLocationAttributeTypesPage manageLocationAttributeTypesPage;
+    private SystemAdministrationPage systemAdministrationPage;
+    private AdministrationPage administrationPage;
+    private static final String LOCATION_ATTRIBUTE_TYPE_NAME = RandomStringUtils.randomAlphabetic(5);
+    private static final String LOCATION_ATTRIBUTE_TYPE_DESCRIPTION = RandomStringUtils.randomAlphabetic(8);
+    private static final String MINIMUM_OCCURS = RandomStringUtils.randomNumeric(2);
+    private static final String MAXIMUM_OCCURS = RandomStringUtils.randomNumeric(5);
+    private static final String DATATYPE_CONFIGURATION = RandomStringUtils.randomAlphabetic(8);
+    private static final String LOCATION_HANDLER_CONFIGURATION = RandomStringUtils.randomAlphabetic(8);
+    private static final String LOCATION_NAME = RandomStringUtils.randomAlphabetic(5);
+    private static final String LOCATION_TAG = RandomStringUtils.randomAlphabetic(5);
+    private static final String LOCATION_DESCRIPTION = RandomStringUtils.randomAlphabetic(8);
+
+    @Before(RunTest.HOOK.SELENIUM_LOCATION_MANAGEMENT)
+    public void visitDashboard() {
+        initiateWithLogin();
+    }
+
+    @After(RunTest.HOOK.SELENIUM_LOCATION_MANAGEMENT)
+    public void destroy() {
+        quit();
+    }
+
+    @Given("a user clicks on the configure metadata link from the home page")
+    public void userClicksOnConfigureMetadataLink() {
+        configureMetadataPage = homePage.goToConfigureMetadata();
+    }
+
+    @Then("the system loads configure metadata link dashboard")
+    public void systemLoadsMetadataDashboard() {
+        assertTrue(textExists("Configure Metadata"));
+    }
+
+    @When("a user clicks on the Manage Location Attribute Types")
+    public void userClicksOnManageLocationAttributeTypes() {
+        manageLocationAttributeTypesPage = configureMetadataPage.goToManageLocationAttributeTypesPage();
+    }
+
+    @Then("the system loads the manage location attribute type page")
+    public void systemLoadsTheManageLocationAttributeTypePage() {
+        assertTrue(textExists("Manage Location Attribute Types"));
+    }
+
+    @And("a user clicks on the add new location attribute type")
+    public void userClicksOnAddNewLocationAttributeType() {
+        manageLocationAttributeTypesPage.goToAddNewLocationAttributeTypeForm();
+    }
+
+    @And("a user fills the add new location attribute type form")
+    public void userFillsAddNewLocationAttributeTypeForm() {
+        manageLocationAttributeTypesPage.enterLocationAttributeTypeName(LOCATION_ATTRIBUTE_TYPE_NAME);
+        manageLocationAttributeTypesPage.enterLocationAttributeTypeDescription(LOCATION_ATTRIBUTE_TYPE_DESCRIPTION);
+        manageLocationAttributeTypesPage.enterMinimumOccurs(MINIMUM_OCCURS);
+        manageLocationAttributeTypesPage.enterMaximumOccurs(MAXIMUM_OCCURS);
+        manageLocationAttributeTypesPage.selectDatatype("Location Datatype");
```

Comment:
Hi @mherman22,
When running this step, sometimes it automatically resets the dropdown selection (datatype) as soon as the value is set. I think it happens if the test starts before the UI is properly loaded. Can you try moving the step `manageLocationAttributeTypesPage.selectDatatype("Location Datatype");` to the bottom?

Reply:
nice catch @Piumal1999 !